test/tests/functional/pbs_resource_usage_log.py

```diff
@@ -0,0 +1,350 @@
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+
+from tests.functional import *
+
+
+class TestResourceUsageLog(TestFunctional):
+
+    """
+    Test various scenarios in which resource usage is logged
+    in the accounting logs
+    """
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+        attr1 = {'job_history_enable': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, attr1)
+        attr2 = {'resources_available.mem': '200gb'}
+        self.server.manager(MGR_CMD_SET, NODE, attr2, id=self.mom.shortname)
+
+    def test_acclog_for_job_states(self):
+        """
+        Check accounting logs when a job completes successfully and when
+        a job is deleted in Q or R state
+        """
+        a = {'Resource_List.select': '1:ncpus=1:mem=200gb'}
+        j1 = Job(TEST_USER, a)
+        j1.create_eatcpu_job(40)
+        jid1 = self.server.submit(j1)
+
+        j2 = Job(TEST_USER, a)
+        j2.create_eatcpu_job(30)
+        jid2 = self.server.submit(j2)
+
+        self.server.expect(JOB, {ATTR_substate: '42'}, jid1)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid2)
+
+        self.server.delete(jid2, wait=True)
+        self.server.expect(JOB, {'job_state': 'F'},
+                           offset=40, extend='x', id=jid1)
+
+        j3 = Job(TEST_USER, a)
+        j3.create_eatcpu_job()
+        jid3 = self.server.submit(j3)
+        self.server.expect(JOB, {ATTR_substate: '42'}, jid3)
+        self.server.delete(jid3, wait=True)
+
+        # No R record ; Only E record for job1 which finished
+        self.server.accounting_match(
+            msg='.*E;' + jid1 +
+            '.*Exit_status=270.*resources_used.*run_count=1', id=jid1,
+            regexp=True)
+        self.server.accounting_match(
+            msg='.*R;' + jid1, id=jid1, existence=False,
```

Comment:
accounting_match() will do a substring match.  You can drop the .* and won't need regexp=True. (and other places in the test)

Reply:
Done